ITensorGaussianMPS/test/gmps.jl

```diff
@@ -123,3 +126,181 @@ end
   @test inner(ψ̃', H, ψ̃) ≈ inner(ψ', H, ψ) rtol = 1e-5
   @test E ≈ energy
 end
+
+@testset "Fermion (BCS real)" begin
+  N = 10
+  Nf = N ÷ 2
+  t = 1.2
+  Delta = 0.5
+  os_h = OpSum()
+  for n in 1:(N - 1)
+    os_h .+= -t, "Cdag", n, "C", n + 1
+    os_h .+= -t, "Cdag", n + 1, "C", n
+  end
+
+  os_p = OpSum()
+  for n in 1:(N - 1)
+    os_p .+= Delta / 2.0, "Cdag", n, "Cdag", n + 1
+    os_p .+= -Delta / 2.0, "Cdag", n + 1, "Cdag", n
+    os_p .+= -Delta / 2.0, "C", n, "C", n + 1
+    os_p .+= Delta / 2.0, "C", n + 1, "C", n
+  end
+
+  h, hb = ITensorGaussianMPS.pairing_hamiltonian(os_h, os_p)
+
+  e, u = eigen(Hermitian(h))
+  #H_D, U_D = Fu.Diag_h(Hermitian(hb))
+  #get correlation matrix
+  E = sum(e[1:N])
+  #@show E
+  # Get the Slater determinant
+  Φ = u[:, 1:N]
+  @test h * Φ ≈ Φ * Diagonal(e[1:N])
+  c = conj(Φ) * transpose(Φ)
+  #c2=Fu.GS_gamma(LinearAlgebra.Diagonal(H_D),U_D)
+  #println("The energy of the ground state is: ", Fu.Energy(c2,(H_D,U_D)))
+  tau = 1.0
+  n, gmps = correlation_matrix_to_gmps(Real.(c); maxblocksize=8, is_bcs=true)
+  ns = round.(Int, n)
+  @test sum(ns) == N
+
+  Λ = c
+  @test gmps * Λ * gmps' ≈ Diagonal(ns) rtol = 1e-2
+  @test gmps' * Diagonal(ns) * gmps ≈ Λ rtol = 1e-2
+
+  # Form the MPS
+  s = siteinds("Fermion", N; conserve_qns=false)
+  psi = correlation_matrix_to_mps(s, c; eigval_cutoff=1e-10, maxblocksize=10)
+
+  # compare entries of the correlation matrix
+  cdagc = correlation_matrix(psi, "Cdag", "C")
+  cdagcdag = correlation_matrix(psi, "Cdag", "Cdag")
+  ccdag = correlation_matrix(psi, "C", "Cdag")
+  cc = correlation_matrix(psi, "C", "C")
+  cblocked = ITensorGaussianMPS.reverse_interleave(c)
+
+  @test all(abs.(cblocked[(N + 1):end, (N + 1):end] - cdagc[:, :]) .< 1e-6)
+  @test all(abs.(cblocked[1:N, 1:N] - ccdag[:, :]) .< 1e-6)
+  @test all(abs.(cblocked[1:N, (N + 1):end] - cc[:, :]) .< 1e-6)
+  @test all(abs.(cblocked[(N + 1):end, 1:N] - cdagcdag[:, :]) .< 1e-6)
+end
+
+@testset "Fermion (BCS real - no pairing)" begin
```

Comment:
I don't think I understand how this test differs from the previous one. I see the title says "No pairing" but much of the code looks similar.

Reply:
The only difference is in the value of `Delta`. There's only minor differences between the three tests, so they could be combined into one if that is prefered. I split them so it's easier to identify what the actual issue is.

Reply:
Probably best to combine them by looping over different values of `Delta`.